tests/unit/index.test.js

```diff
@@ -1,5 +1,27 @@
-import { describe, expect, it } from "vitest";
+import { describe, expect, it, test } from "vitest";
+import { shortAddr } from '/src/shared/lib/utils';  // Error from Vitest: ReferenceError: window is not defined
 
+
+// Unit-tests to verify "shortAddr(addr)" function accepts only a string as "addr" argument
+describe("shortAddr", () => {
+	it("Positive test cases - if 'addr' argument is a string", () => {
+		expect(shortAddr("string")).not.toThrow();
+		expect(shortAddr("234")).not.toThrow();
+		expect(shortAddr("")).not.toThrow();
+	});
+	it("Negative test cases - throw an error if 'addr' argument is NOT a string", () => {
+		expect(() => shortAddr(345)).toThrow();
+		expect(() => shortAddr(true)).toThrow();
+		expect(() => shortAddr(null)).toThrow();
+		expect(() => shortAddr(undefined)).toThrow();
+		expect(() => shortAddr([])).toThrow();
```

Comment:
Usually, it’s not necessary to test all JavaScript types. By the way, such errors are typically caught using TypeScript/JSDoc. I would suggest considering this approach in the future

Reply:
Got you, thanks.

Reply:
I mean, it’s not really necessary here to test for every JavaScript type unless you expect them to be passed to your function at runtime.

And such JSDoc comment will solve most of such issues during development without event using TS explicitly (but it's TS under the hood):
```
/**
 * Shortens an address string to a more readable format by showing only the first 6 characters,
 * followed by an ellipsis, and then the last 4 characters.
 *
 * @param {string} addr - The address string to shorten.
 * @returns {string} The shortened address string.
 * @throws {TypeError} If the `addr` argument is not a string.
 */
export function shortAddr(addr) {
  if (typeof addr !== "string") {
    throw new TypeError("'addr' argument must be a string");
  }
  return addr.substring(0, 6) + "..." + addr.substring(addr.length - 4);
}
```
Check it in your editor. It might pick up such syntax by default and prevent you from accidentally passing incorrect types to the function. You can learn more here: https://dev.to/samuel-braun/boost-your-javascript-with-jsdoc-typing-3hb3